akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -304,14 +305,21 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
       }
 
       def reachabilityChanged(keysForNode: Set[AbstractServiceKey], newState: State): Unit = {
-        keysForNode.foreach { changedKey =>
+        notifySubscribers(keysForNode, servicesWereAddedOrRemoved = false, newState)
```

Comment:
I have to check what servicesWereAddedOrRemoved is used for, but this is how it was before. Reachable/Unreachable is not changing allServiceInstances.

Reply:
The theoretically possible scenario would be:
* ddata update seen first, entry added to the registry but corresponding node not seen yet
* UnreachableMember arrives before MemberJoined, the unreachable can be for any node not just for the joining

Hm, this is difficult to keep in sync. I was thinking if we could keep track of previously published Listing and compare, but it's for each key and also subscribers can be added after the previously published Listing.

This is like an optimization hint and we can error on the side of servicesWereAddedOrRemoved=true even though it was the same allServiceInstances as the previous. Maybe we can use that in some way. Have to think some more about it.

Reply:
Second thought, that scenario isn't a problem. It will publish a Listing with servicesWereAddedOrRemoved=true when the MemberJoined arrives. I don't think this is a problem.